app/api/business/insight_business.py

```diff
@@ -0,0 +1,39 @@
+import pendulum
+from app.api.services import insight_service
+from app.api.business.errors import NotFoundError
+
+
+def get_insight(current_user, now):
+    role = current_user.role if hasattr(current_user, 'role') else None
+    if role == 'admin':
+        insight = insight_service.get_insight(now, False)
+    else:
+        insight = insight_service.get_insight(now)
+
+    if not insight:
+        raise NotFoundError("Invalid date '{}'".format(now))
+
+    return insight.get('data')
+
+
+def upsert(now, data, active):
+    existing = insight_service.get_insight_for_update(now)
+    saved = None
+    if existing:
+        if data is not None:
```

Comment:
```suggestion
        if data:
```
Does this need to check for `None` specifically or just whether it is truthy?

Reply:
needs to check for None specifically as it should only do something when it is not None.

Updated function with default of None.